delta-lake/src/integration/java/org/apache/iceberg/delta/TestSnapshotDeltaLakeTable.java

```diff
@@ -0,0 +1,468 @@
+    externalDataFilesIdentifier = destName(defaultSparkCatalog, externalDataFilesTableName);
+
+    spark.sql(String.format("DROP TABLE IF EXISTS %s", partitionedIdentifier));
+    spark.sql(String.format("DROP TABLE IF EXISTS %s", unpartitionedIdentifier));
+    spark.sql(String.format("DROP TABLE IF EXISTS %s", externalDataFilesIdentifier));
+
+    // hard code the dataframe
+    List<String> jsonList = Lists.newArrayList();
+    jsonList.add(row1);
+    jsonList.add(row2);
+    jsonList.add(row3);
+    jsonList.add(row4);
+    jsonList.add(row5);
+    JavaSparkContext javaSparkContext = JavaSparkContext.fromSparkContext(spark.sparkContext());
+    SQLContext sqlContext = new SQLContext(javaSparkContext);
+    JavaRDD<String> rdd = javaSparkContext.parallelize(jsonList);
+    Dataset<Row> df = sqlContext.read().json(rdd);
+
+    // write to delta tables
+    df.write()
+        .format("delta")
+        .mode(SaveMode.Append)
+        .partitionBy("id")
+        .option("path", partitionedLocation)
+        .saveAsTable(partitionedIdentifier);
+
+    df.write()
+        .format("delta")
+        .mode(SaveMode.Append)
+        .option("path", unpartitionedLocation)
+        .saveAsTable(unpartitionedIdentifier);
+
+    df.write()
+        .format("delta")
+        .mode(SaveMode.Append)
+        .option("path", externalDataFilesTableLocation)
+        .saveAsTable(externalDataFilesIdentifier);
+
+    // Delete a record from the table
+    spark.sql("DELETE FROM " + partitionedIdentifier + " WHERE id=3");
+    spark.sql("DELETE FROM " + unpartitionedIdentifier + " WHERE id=3");
+
+    // Update a record
+    spark.sql("UPDATE " + partitionedIdentifier + " SET id=3 WHERE id=1");
+    spark.sql("UPDATE " + unpartitionedIdentifier + " SET id=3 WHERE id=1");
+  }
+
+  @After
+  public void after() {
+    // Drop delta lake tables.
+    spark.sql(
+        String.format(
+            "DROP TABLE IF EXISTS %s", destName(defaultSparkCatalog, partitionedTableName)));
+    spark.sql(
+        String.format(
+            "DROP TABLE IF EXISTS %s", destName(defaultSparkCatalog, unpartitionedTableName)));
+    spark.sql(
+        String.format(
+            "DROP TABLE IF EXISTS %s", destName(defaultSparkCatalog, externalDataFilesTableName)));
+
+    spark.sql(String.format("DROP DATABASE IF EXISTS %s", NAMESPACE));
+  }
+
+  @Test
+  public void testBasicSnapshotPartitioned() {
+    String newTableIdentifier = destName(icebergCatalogName, "iceberg_table");
+    SnapshotDeltaLakeTable.Result result =
+        DeltaLakeToIcebergMigrationSparkIntegration.snapshotDeltaLakeTable(
+                spark, newTableIdentifier, partitionedLocation)
+            .execute();
+
+    checkSnapshotIntegrity(partitionedLocation, partitionedIdentifier, newTableIdentifier, result);
+    checkIcebergTableLocation(newTableIdentifier, partitionedLocation);
+    spark.sql(String.format("DROP TABLE IF EXISTS %s", newTableIdentifier));
```

Comment:
if the test ever fails, then this SQL won't be executed

Reply:
Thank you for pointing this out. I moved them to `after()`